src/dmd/cppmanglewin.d

```diff
@@ -684,22 +684,138 @@ private:
             buf.writestring("?1");
             return;
         }
+        else if (sym.ident)
+        {
+            switch (sym.ident.toString)
+            {
+                case "opAssign":    buf.writestring("?4");  return;
+                case "opEquals":    buf.writestring("?8");  return;
+                case "opIndex":     buf.writestring("?A");  return;
+                case "opCall":      buf.writestring("?R");  return;
+                default:            break;
+            }
+        }
         if (TemplateInstance ti = sym.isTemplateInstance())
         {
+            auto id = ti.tempdecl.ident;
+            const(char)* ident = id.toChars;
+
+            // test for special symbols
+            int firstTemplateArg = 0;
+            int whichOp = -1;
+            switch (ti.name.toString)
```

Comment:
Resorting to string compares is terribly inefficient. DMD puts all identifiers into a hash table, and then comparing identifiers becomes a simple comparison of pointers.